samples/OpenIdConnect.AzureSample/Startup.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Linq;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Authentication;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+
+namespace OpenIdConnect.AzureSample
+{
+    public class Startup
+    {
+        private const string GraphResourceID = "https://graph.windows.net";
+
+        public Startup()
+        {
+            Configuration = new ConfigurationBuilder()
+                .AddEnvironmentVariables()
+                .AddUserSecrets()
+                .Build();
+        }
+
+        public IConfiguration Configuration { get; set; }
+
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddAuthentication(sharedOptions =>
+                sharedOptions.SignInScheme = CookieAuthenticationDefaults.AuthenticationScheme);
+        }
+
+        public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
+        {
+            loggerfactory.AddConsole(LogLevel.Information);
+
+            // Simple error page
+            app.Use(async (context, next) =>
+            {
+                try
+                {
+                    await next();
+                }
+                catch (Exception ex)
+                {
+                    if (!context.Response.HasStarted)
+                    {
+                        context.Response.Clear();
+                        context.Response.StatusCode = 500;
+                        await context.Response.WriteAsync(ex.ToString());
+                    }
+                    else
+                    {
+                        throw;
+                    }
+                }
+            });
+
+            app.UseIISPlatformHandler();
+
+            app.UseCookieAuthentication(new CookieAuthenticationOptions
+            {
+                AutomaticAuthenticate = true
+            });
+
+            var clientId = Configuration["oidc:clientid"];
+            var clientSecret = Configuration["oidc:clientsecret"];
+            var authority = Configuration["oidc:authority"];
+            var resource = "https://graph.windows.net";
+            app.UseOpenIdConnectAuthentication(new OpenIdConnectOptions
+            {
+                ClientId = clientId,
+                ClientSecret = clientSecret, // for code flow
```

Comment:
https://github.com/aspnet/Security/issues/701 opened to cover Certificates and Signed Assertions.